build/build.ps1

```diff
@@ -21,14 +21,14 @@ function Build-One {
     );
 
     Write-Host "##[info]Building $project ($action)..."
-    dotnet $action (Join-Path $PSScriptRoot $project) `
-        -c $Env:BUILD_CONFIGURATION `
-        -v $Env:BUILD_VERBOSITY `
-        /property:DefineConstants=$Env:ASSEMBLY_CONSTANTS `
-        /property:Version=$Env:ASSEMBLY_VERSION `
-        /property:QsharpDocsOutputPath=$Env:DOCS_OUTDIR
-
-    if  ($LastExitCode -ne 0) {
+    Invoke-Expression ("dotnet $action $(Join-Path $PSScriptRoot $project) " +
+        "-c $Env:BUILD_CONFIGURATION " +
+        "-v $Env:BUILD_VERBOSITY " +
+        "$(if ($Env:ASSEMBLY_CONSTANTS) { "/property:DefineConstants=$Env:ASSEMBLY_CONSTANTS" }) " +
+        "/property:Version=$Env:ASSEMBLY_VERSION " +
+        "/property:QsharpDocsOutputPath=$Env:DOCS_OUTDIR")
```

Comment:
It is often easier to use PowerShell's argument splat operators to expand a variable list of arguments than to use `Invoke-Expression`. There's some subtle bugs that can be introduced by `Invoke-Expression` swallowing `LASTEXITCODE` that often make spalatting a simpler solution.
```suggestion
    if ("" -ne "$Env:ASSEMBLY_CONSTANTS") {
        $args = @("/property:DefineConstants=$Env:ASSEMBLY_CONSTANTS");
    }  else {
        $args = @();
    }
    dotnet $action $(Join-Path $PSScriptRoot $project) `
        -c $Env:BUILD_CONFIGURATION `
        -v $Env:BUILD_VERBOSITY  `
        @args `
        /property:Version=$Env:ASSEMBLY_VERSION `
        /property:QsharpDocsOutputPath=$Env:DOCS_OUTDIR;
```
Using `@args` instead of `$args` there expands the contents of `$args` as a list of arguments; if `$args` is empty, then `@args` is a no-op.

Reply:
Oh, neat! I'll have to try that instead.